src/timecop.cr

```diff
@@ -4,96 +4,107 @@ require "./timecop/core_ext/*"
+
   @@stack = [] of TimeStackItem
-  @@safe_mode = false
 
-  # Returns whether or not Timecop is currently frozen/travelled/scaled
-  def frozen?
+  # Returns whether or not `Timecop` is currently frozen/travelled/scaled
+  def frozen? : Bool
     !@@stack.empty?
   end
 
-  # Allows you to run a block of code and "fake" a time throughout the execution of that block.
-  def freeze(time : Time)
+  # Allows you to run a block of code and "fake" a time throughout
+  # the execution of that block.
+  def freeze(time : Time) : Time
     send_travel(:freeze, time)
   end
 
-  # Allows you to run a block of code and "fake" a time throughout the execution of that block.
-  def freeze(time : Time, &block : Time -> )
-    send_travel(:freeze, time, &block)
+  # :ditto:
+  def freeze(time : Time, & : Time -> V) : V forall V
+    send_travel(:freeze, time) { |t| yield t }
   end
 
-  # Allows you to run a block of code and "fake" a time throughout the execution of that block.
-  def travel(time : Time)
+  # :ditto:
+  def travel(time : Time) : Time
     send_travel(:travel, time)
   end
 
-  def travel(time : Time, &block : Time -> )
-    send_travel(:travel, time, &block)
+  # :ditto:
+  def travel(time : Time, & : Time -> V) : V forall V
+    send_travel(:travel, time) { |t| yield t }
   end
 
-  # Allows you to run a block of code and "scale" a time throughout the execution of that block.
-  # The first argument is a scaling factor, for example:
-  #   Timecop.scale(2) do
-  #     ... time will 'go' twice as fast here
-  #   end
+  # Allows you to run a block of code and "scale" a time throughout
+  # the execution of that block.
+  #
+  # ```
+  # Timecop.scale(2) do
+  #   # ... time will 'go' twice as fast here
+  # end
+  # ```
+  #
   # Returns the value of the block if one is given, or the mocked time.
-  def scale(time : Time, factor : Float64)
-    send_travel(:scale, time, factor)
+  def scale(factor : Float64) : Time
+    send_travel(:scale, factor)
   end
 
-  def scale(time : Time, factor : Float64, &block : Time -> )
-    send_travel(:scale, time, factor, &block)
+  # :ditto:
+  def scale(time : Time, factor : Float64) : Time
+    send_travel(:scale, time, factor)
   end
 
-  # Reverts back to system's Time.now
-  def return
-    unmock!
+  # :ditto:
+  def scale(factor : Float64, & : Time -> V) : V forall V
+    send_travel(:scale, factor) { |t| yield t }
   end
 
-  # Reverts back to system's Time.now
-  def return(&block)
-    stack_backup = @@stack.dup
-    unmock!
-    yield
-  ensure
-    @@stack = stack_backup.as(Array(Timecop::TimeStackItem))
+  # :ditto:
+  def scale(time : Time, factor : Float64, & : Time -> V) forall V
+    send_travel(:scale, time, factor) { |t| yield t }
   end
 
-  def safe_mode=(safe)
-    @@safe_mode = safe
+  # Reverts back to system's `Time.local`
+  def return : Nil
+    unmock!
   end
 
-  def safe_mode?
-    @@safe_mode
+  # :ditto:
+  def return(& : -> V) : V forall V
```

Comment:
I don’t see the reason to add explicit `V forall V` to all these block return types

Reply:
I do. The explicitness _is_ the reason.